prov/shm/src/smr_ep.c

```diff
@@ -170,7 +171,7 @@ void smr_progress_resp(struct smr_ep *ep)
 		resp = ofi_cirque_head(smr_resp_queue(ep->region));
 		if (resp->status)
 			break;
-		ep->tx_comp(ep, NULL);
+		ep->tx_comp(ep, NULL, resp->flags);
```

Comment:
The completion flags should come from something stored locally, rather than the peer returning it, if possible.  Where does the void *context associated with the transmit call come from?